### PR TITLE
refactor: pass checkout state as criterion to findBy method

### PR DESCRIPTION
**Pull request by EmilMassey**

### CODE REVIEW

File: src/ViewRepository/Order/PlacedOrderViewRepository.php [hunk 1]
```diff
@@ -44,10 +44,8 @@ public function getAllCompletedByCustomerEmail(string $customerEmail): array
         $cartViews = [];
 
         /** @var OrderInterface $order */
-        foreach ($this->orderRepository->findBy(['customer' => $customer]) as $order) {
-            if ($order->getCheckoutState() === OrderCheckoutStates::STATE_COMPLETED) {
-                $cartViews[] = $this->placedOrderViewFactory->create($order, $order->getLocaleCode());
-            }
+        foreach ($this->orderRepository->findBy(['customer' => $customer, 'checkoutState' => OrderCheckoutStates::STATE_COMPLETED]) as $order) {
```
Comment (lchrusciel) [inline src/ViewRepository/Order/PlacedOrderViewRepository.php]: We should list all orders that are not in `cart` state. There should be repository method for it already.

Comment (mamazu) [reply src/ViewRepository/Order/PlacedOrderViewRepository.php]: Yes there is the `findByCustomer` method.

**Comment by mamazu** [comment]
I have rebased the branch to the current master and refactored the method in the order repository. Now it should be ready to be merged.

> The base of this pull-request was changed, you need fetch and reset your local branch
if you want to add new commits to this pull request. Reset before you pull, else commits
may become messed-up. Unless you added new commits (to this branch) locally that you did not push yet,
execute `git fetch origin && git reset "repository-refactor"` to update your local branch. Feel free to ask for assistance when you get stuck. :+1:

**Comment by lchrusciel** [comment]
Thank you, Emil! :1st_place_medal:

**Comment by EmilMassey** [comment]
Thank you for your help!

**Pull request merged**